Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -201,130 +207,130 @@ struct SymbolGraphLoader {
-                        if !symbolAvailability.contains(fallbackPlatform) {
-                            for var fallbackAvailability in symbolAvailability.availability {
-                                // Add the platform fallback to the availability mixin the platform is inheriting from.
-                                // The added availability copies the entire availability information,
-                                // including deprecated and obsolete versions.
-                                if fallbackAvailability.matches(inheritedPlatform) {
-                                    fallbackAvailability.domain = SymbolGraph.Symbol.Availability.Domain(rawValue: fallbackPlatform.rawValue)
-                                    symbolAvailability.availability.append(fallbackAvailability)
-                                }
+                }
+                
+                // Add the module default availability information.
+                defaultAvailabilities.forEach { defaultAvailability in
+                    // Check that the symbol not has explicit availability for this platform already,
+                    // Check that if there was a symbolgraph for this platform, the symbol was present on it,
+                    // if not it means that the symbol is not available for the default platform.
+                    if (
+                        !platformsAvailableAtThisSymbol.contains(defaultAvailability.platformName.rawValue) &&
+                        (platformsRegistered[defaultAvailability.platformName.rawValue]?.contains(where: { symbolID == $0.precise }) ?? true)
+                    ) {
+                        // If the  missing availability corresponds to a fallback platform, and there's default availability for the platform that this one fallbacks from, don't add it.
+                        if let fallbackPlatform = fallbackPlatforms.first(where: { $0.key == defaultAvailability.platformName }), platformsAvailableAtThisSymbol.contains(fallbackPlatform.value.rawValue) {
+                            return
+                        }
+                        guard var defaultAvailabilityItem = AvailabilityItem(defaultAvailability) else { return }
+                        
+                        // Look if the symbol already has this availability item.
+                        if let idx = symbolAvailability.availability.firstIndex(where: {
+                            $0.domain?.rawValue == defaultAvailability.platformName.rawValue
+                        }) {
+                            // If the symbol is marked as unavailable don't add the default availability.
+                            if  symbolAvailability.availability[idx].isUnconditionallyUnavailable {
+                                return
                             }
+                            defaultAvailabilityItem.deprecatedVersion = symbolAvailability.availability[idx].deprecatedVersion
```

Comment:
Question: Is it only the deprecated version that we want to move over here (if so; why?) or do we also care about `obsoletedVersion`, `renamed`, `isUnconditionallyDeprecated`, etc.?

Reply:
Thanks for pointing this out. `renamed` does not surface the docs so there's no need to copy it. If it's obsoleted then we should not display the item so we should return earlier. `isUnconditionallyDeprecated` is the only one that should be copied over.